rjc_account_invoice_reimbursable/models/account_invoice.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2019 Ecosoft Co., Ltd (http://ecosoft.co.th/)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+from odoo import api, models, fields
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    @api.model
+    def line_get_convert(self, line, part):
+        val = self.env['product.product']._convert_prepared_anglosaxon_line(
+            line, part)
+        for record in self.tax_line_ids:
+            if line['type'] == 'tax' and record.sequence != 1 and \
+                    line['invoice_tax_line_id'] == record.id:
+                reimburs_tax = self.reimbursable_ids.filtered(
+                    lambda l: l.sequence == record.sequence)
+                part = self.env['res.partner']._find_accounting_partner(
+                    reimburs_tax.partner_id)
+                val['partner_id'] = part.id
+        return val
+
+    @api.onchange('invoice_line_ids', 'reimbursable_ids')
+    def _onchange_invoice_line_reimbursable_ids(self):
```

Comment:
Per say, I think you can use self._context.get('reimbursable_ids') to get latest sequence on the Form, and get the next one.